apps/sms/lib/bridge/plugins/dispose/client.js

```diff
@@ -0,0 +1,59 @@
+!((define)=>{define((require,exports,module)=>{
+
+  /**
+   * Logger
+   * @type {Function}
+   * @const
+   */
+  const debug = 1 ? function(arg1, ...args) {
+    var type = `[${self.constructor.name}][${location.pathname}]`;
+    console.log(`[DisposableClient]${type} - "${arg1}"`, ...args);
+  } : () => {};
+
+  /**
+   * Exports
+   */
+  module.exports = (client) => {
+    var pendingOperationManager = {
+      operations: new Set(),
+      registerOperation(operation) {
+        operation = operation.then(
+          () => this.operations.delete(operation),
+          () => this.operations.delete(operation)
+        );
+        this.operations.add(operation);
+      }
```

Comment:
with this, we need to take care that all operations do not last longer than their promises. I'm thinking that we cold have an operation that takes longer (using events for example) than their promise resolution. Quite hypothetic right now but I don't think it's that uncommon.